actions/class.Previewer.php

```diff
@@ -0,0 +1,225 @@
+                    break;
+
+                case $e instanceof \tao_models_classes_FileNotFoundException:
+                    $code = 404;
+                    break;
+            }
+        }
+        return $code;
+    }
+
+    /**
+     * Initializes the delivery session
+     */
+    public function init()
+    {
+        $code = 200;
+
+        try {
+
+            $serviceCallId = $this->getRequestParameter('serviceCallId');
+
+            $response = [
+                'success' => $serviceCallId == 'previewer',
+                'itemIdentifier' => null,
+                'itemData' => null
+            ];
+
+        } catch (\Exception $e) {
+            $response = $this->getErrorResponse($e);
+            $code = $this->getErrorCode($e);
+        }
+
+        $this->returnJson($response, $code);
+    }
+
+    /**
+     * Provides the definition data and the state for a particular item
+     */
+    public function getItem()
+    {
+        $code = 200;
+
+        try {
+            $itemUri = $this->getRequestParameter('itemUri');
+            $resultId = $this->getRequestParameter('resultId');
+
+            $response = [
+                'baseUrl' => '',
+                'content' => [],
+            ];
+
+            // previewing a result
+            if ($resultId) {
+                if (!$this->hasRequestParameter('itemDefinition')) {
+                    throw new \common_exception_MissingParameter('itemDefinition', $this->getRequestURI());
+                }
+
+                if (!$this->hasRequestParameter('deliveryUri')) {
+                    throw new \common_exception_MissingParameter('deliveryUri', $this->getRequestURI());
+                }
+
+                $itemDefinition = $this->getRequestParameter('itemDefinition');
+                $delivery = new \core_kernel_classes_Resource($this->getRequestParameter('deliveryUri'));
+
+                $itemPreviewer = new \oat\taoQtiTest\models\ItemPreviewer();
+                $itemPreviewer->setServiceLocator($this->getServiceLocator());
+
+                $response['content'] = $itemPreviewer->setResultId($resultId)
+                    ->setItemDefinition($itemDefinition)
+                    ->setDelivery($delivery)
+                    ->loadCompiledItemData();
+
+                $response['baseUrl'] = $itemPreviewer->getBaseUrl();
+
+            } else if ($itemUri) {
+                // Load RESOURCE item data
+                // TODO
+            } else {
+                throw new \common_exception_BadRequest('Either itemUri or resultId needs to be provided.');
+            }
+
+            $response['success'] = true;
+        } catch (\Exception $e) {
+            $response = $this->getErrorResponse($e);
+            $code = $this->getErrorCode($e);
+        }
+
+        $this->returnJson($response, $code);
+    }
+
+    /**
+     * Stores the state object and the response set of a particular item
+     */
+    public function submitItem()
```

Comment:
Is it needed for the previewer ?

Reply:
For the time being no, we can remove it.